File: src/main/java/com/github/wakingrufus/mastodon/events/OAuthStartEvent.java
package com.github.wakingrufus.mastodon.events;

import com.sys1yagi.mastodon4j.MastodonClient;
import com.sys1yagi.mastodon4j.api.entity.auth.AppRegistration;
import javafx.beans.NamedArg;
import javafx.event.Event;
import javafx.event.EventTarget;
import javafx.event.EventType;
import lombok.Getter;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Getter
public class OAuthStartEvent extends Event {
    public static final EventType<OAuthStartEvent> OAUTH_START = new EventType<>(Event.ANY, "OAUTH_START");


    private AppRegistration appRegistration;
    MastodonClient client;

    public OAuthStartEvent(@NamedArg("source") Object source,
                           @NamedArg("target") EventTarget target,
                           AppRegistration appRegistration,  MastodonClient client) {
        super(source, target, OAUTH_START);
        this.client = client;
        this.appRegistration = appRegistration;
    }

}

File: src/main/kotlin/com/github/wakingrufus/mastodon/events/OAuthStartEvent.kt
package com.github.wakingrufus.mastodon.events

import com.sys1yagi.mastodon4j.MastodonClient
import com.sys1yagi.mastodon4j.api.entity.auth.AppRegistration
import javafx.beans.NamedArg
import javafx.event.Event
import javafx.event.EventTarget
import javafx.event.EventType

val OAUTH_START: EventType<OAuthStartEvent> = EventType<OAuthStartEvent>(Event.ANY, "OAUTH_START")

class OAuthStartEvent(@NamedArg("source") source: Any,
                      @NamedArg("target") target: EventTarget,
                      val appRegistration: AppRegistration,
                      val client: MastodonClient)
    : Event(source, target, OAUTH_START) {

}
